src/evaluators/ForStatement.js

```diff
@@ -433,8 +433,7 @@ function generateRuntimeForStatement(
       "PP0037",
       "FatalError"
     );
-    realm.handleError(diagnostic);
-    throw new FatalError();
+    if (realm.handleError(diagnostic) !== "Recover") throw new FatalError();
```

Comment:
If we just carry on from this point, we'll do so with an incorrect state, which may led to weird and misleading errors later on in the compilation.

Reply:
It would only carry on in an environment where handling a FatalError returned "Recover", which was only happening in tests.